www/src/components/CodeBlock.tsx

```diff
@@ -32,12 +32,30 @@ export type CollapsibleLiveEditorTypes = {
 
 function CollapsibleLiveEditor({ children }: CollapsibleLiveEditorTypes) {
   const [collapseIsOpen, setCollapseIsOpen] = useState(false);
+
+  const eventSegmentSubmitter = (e: Event & { target: HTMLElement }) => {
+    const useCaseElement: { parentNode: any } = e.target.closest('.pgn-doc__code-block') as HTMLElement;
+    let elementBeforeUseCase: {
+      className: string, id: string, previousSibling: any
+    } = useCaseElement.parentNode.previousSibling;
+    const componentNameAndCategory: string = window.location.pathname.replace(/\//g, '.');
+
+    while (elementBeforeUseCase.className !== 'pgn-doc__heading') {
+      elementBeforeUseCase = elementBeforeUseCase.previousSibling;
+    }
```

Comment:
can we move this part into separate function? otherwise it's not clear what is going on here, you could name the function `getCodeBlockHeading` and everything would become much clearer for other engineers

Reply:
Done, thanks